go/vt/vttablet/tabletserver/gc/tablegc.go

```diff
@@ -67,6 +67,7 @@ func registerGCFlags(fs *pflag.FlagSet) {
 	fs.DurationVar(&purgeReentranceInterval, "gc_purge_check_interval", purgeReentranceInterval, "Interval between purge discovery checks")
 	// gcLifecycle is the sequence of steps the table goes through in the process of getting dropped
 	fs.StringVar(&gcLifecycle, "table_gc_lifecycle", gcLifecycle, "States for a DROP TABLE garbage collection cycle. Default is 'hold,purge,evac,drop', use any subset ('drop' implcitly always included)")
+	fs.DurationVar(&leaderCheckInterval, "leader_check_interval", leaderCheckInterval, "Interval between transition to primary is check.")
```

Comment:
I don't exactly what we are checking at this interval, but we need to improve the description.
"Interval at which we check whether the primary has changed"?? I don't know, we need to find out.
@shlomi-noach any objection to making this a flag?

Reply:
Looks good to me and no objection creating the flag. I recommend renaming it as `tablegc_leader_check_interval`.
Taking another look, I think this entire variable can be altogether removed with some refactoring. The check for leader-or-not-leader really means primary-on-not-primary, which does have to be _checked_ per se, the resolution can be made via tablet's state_manager. I can look into that. So I'm not sure whether you'd want to add a variable that will be deprecated shortly afterwards?

A similar variable exists in the tablet throttler, but in the throttler case we won't be able to use state_manager.

Reply:
I was eager to make the change. Please consider https://github.com/vitessio/vitess/pull/11972. I only expect https://github.com/vitessio/vitess/pull/11972 to be merged post-Holidays. In the meantime, I have no objection to adding the configurable variable -- just note that it will be removed shortly afterwards...

Reply:
@rsajwani given this let us not add a flag here.